asserts/repair.go

```diff
@@ -162,25 +162,33 @@ func assembleRepair(assert assertionBase) (Assertion, error) {
 		return nil, err
 	}
 
-	// verify that modes is a list of only "run", "recover" and "install"
+	// validate that all base snap names are valid snap names
+	for _, b := range bases {
+		if err := naming.ValidateSnap(b); err != nil {
+			return nil, fmt.Errorf("cannot use %q as element in \"bases\": %v", b, err)
+		}
+	}
+
+	// verify that modes is a list of only "run" and "recover"
 	if len(modes) != 0 {
 		for _, m := range modes {
 			// note we could import boot here to use i.e. boot.ModeRun, but that
```

Comment:
we definitely don't want to import boot from asserts :)  (asserts is definitely used outside of snapd so its deps should stay lean and minimal and not too entangled with other parts)